### DISCUSSION TITLE
Does `.insert` reset the TTL time?

### DISCUSSION THREAD

**Discussion by tqwewe** [moka-rs/moka#410] [Q&A] [answered]
If I call `.insert` on an item that already exists, will this item's TTL still be updated to the current time + ttl? Or will it still evict this item based on the initial `.insert` call when the item wasn't present in the cache?

**Comment by tatsuya6502** [comment, marked as answer]
Thank you for using moka.

> If I call `.insert` on an item that already exists, will this item's TTL still be updated to the current time + ttl?

Yes, the item's TTL must be updated.

Here is an unit test: https://github.com/moka-rs/moka/blob/v0.12.5/src/sync/cache.rs#L2470-L2491

1. Create a cache with TTL 10 secs.
2. Insert a key "b" with value "bob".
3. Wait for 5 secs.
4. Insert the same key "b" with value "bill".
5. Wait for 5 secs.
6. Get the key "b".
    - → "b" with "bill" still exists. (10 secs past from the initial insertion, but only 5 secs past from the update)

```rust
        let mut cache = Cache::builder()
            .max_capacity(100)
            .time_to_live(Duration::from_secs(10))
            .eviction_listener(listener)
            .build();

        ...

        cache.insert("b", "bob");
        cache.run_pending_tasks();

        assert_eq!(cache.entry_count(), 1);

        mock.increment(Duration::from_secs(5)); // 15 secs.
        cache.run_pending_tasks();

        assert_eq!(cache.get(&"b"), Some("bob"));
        assert!(cache.contains_key(&"b"));
        assert_eq!(cache.entry_count(), 1);

        cache.insert("b", "bill");
        expected.push((Arc::new("b"), "bob", RemovalCause::Replaced));
        cache.run_pending_tasks();

        mock.increment(Duration::from_secs(5)); // 20 secs
        cache.run_pending_tasks();

        assert_eq!(cache.get(&"b"), Some("bill"));
        assert!(cache.contains_key(&"b"));
        assert_eq!(cache.entry_count(), 1);
```
